### DISCUSSION TITLE
pKa calculation with FDPB(mg-auto) vs TABI-PB

### DISCUSSION THREAD

**Discussion by rwxayheee** [Electrostatics/apbs#189] [Q&A] [answered]
Dear APBS experts, 

I’m trying to calculate the intrinsic pKa of a residue in protein using APBS. I found [a tutorial for examples/bem-pKa](https://dasher.wustl.edu/chem478/labs/lab-08/tutorial-pka.pdf) but that version is a bit old. It uses FDPB (mg-auto) probably because it was made before the incorporation of TABI. I also read the current version of examples/bem-pKa, which uses the TABI-PB solver, and I have some questions regarding the difference between the FDPB and the TABI version: 

1. Choice of “sdens”: In the FDPB version, we are using sdens = 40 for the FDPB (mg-auto). This is probably to reduce the discretization errors for the calculation. As said [in the current documentation](https://apbs.readthedocs.io/en/latest/using/input/old/generic/sdens.html?highlight=sdens), a typical value is 10 for general purposes. 
But from the sample input files of the current TABI version, we are using sdens = 10 for a standalone Asp residue, and sdens = 2.47 for the entire lysosome. Is this a feature of TABI, or is it just fine to use a default value of 10 for all steps? 

2. The thermodynamic cycle: Is it normal to get the same electrostatic energies from “2lzt-noASH66” and “2lzt-noASP66” (protein containing the uncharged residue)? If the results are usually the same, could we skip this calculation? 

3. Do we need to calculate the electrostatic energy of a dissociated proton? Or is it usual to assume that the energy of a proton is just the same, in solvent and in the protein environment?

Any comments/suggestions would be really appreciated. Thank you for your time and kind advice in advance. 

Massive Thanks,
Amy

**Comment by sobolevnrm** [comment]
Hello --

1. I'm not as familiar with `sdens` behavior in TABI.  @lwwilson1 can you help here?
2. If the uncharged states are the reference state for each, then they should be identical.
3. The proton reference state should be equivalent to the activity of the proton (e.g., the pH).

Thanks

**Reply by lwwilson1** [reply]
Hi there, sorry for the delay, I should turn GitHub notifications on…

I believe the choice of 2.47 was simply to optimize the calculation time of the full complexes and provide the same resolution as the standalone residue. Using 10 for all should be fine, though the calculation will take quite a bit longer. 

I’m glad you pointed out these files however, as they’re a bit misleading. After the rollover to APBS 3, we deprecated MSMS support in favor of NanoShaper for TABI-PB. So `sdens` isn’t actually the density in 1/A^2, but scales closer to 1/A, where the denominator is the side length of cube. I’ll make sure these and the docs get updated. Please reach out if you have any questions about TABI-PB usage.

**Reply by rwxayheee** [reply]
Hi Nathan and Leighton, Thank you for your kind replies! I have one more question about the output file. Here's an example: 

> *** OUTPUT FOR TABI-PB RUN ***
> 
>     Solvation energy = -2.790511 kJ/mol
>          Free energy = -82.547106 kJ/mol
> 
> The max and min potential and normal derivatives on vertices:
>         Potential min: -3.548462, max: 4.809647
> Normal derivative min: -19.801658, max: 15.597001
> 
> 
> 
> 
> Returning to APBS caller...
> 
> Solvation energy and Coulombic energy in kJ/mol...
> 
>   Global net ELEC energy = -2.790511328181E+00
>   Global net COULOMBIC energy = -7.975659516512E+01
> 

If I do "mg-auto" calculation with "calcenergy total", APBS will return just one Total electrostatic energy. Here from the "tabi" calculation I got several outputs. And the "Free energy = -82.547106 kJ/mol" is the total apolar energy of the system, and we should use it for pKa calculation, is my understanding correct? It seems to be a sum of "Global net ELEC energy" and "Global net COULOMBIC energy". 

I'm still learning to use APBS and TABI. Any comments/suggestions would be really appreciated. Thank you again for your time and kind support!

**Comment by sobolevnrm** [comment]
@lwwilson1 would you be able to help answer this question?

**Comment by lwwilson1** [comment, marked as answer]
Yes, that’s correct; free energy is the quantity to use here. I’ll note that for TABI that’s purely polar energy; no non-polar components are included there.

**Reply by rwxayheee** [reply]
Got it! Thank you again for your kind reply!!
